pkg/broker/filter/filter_handler.go

```diff
@@ -180,7 +180,9 @@ func (r *Handler) serveHTTP(ctx context.Context, event cloudevents.Event, resp *
 		// framework returns a 500 to the caller, so the Channel would send this repeatedly.
 		return nil
```

Comment:
Notes for the future: this should return a 400 rather than silently dropping with a 202. Then whatever upstream is sending the event (likely the channel) can handle the error as configured (for example, putting the event in a DLQ).

Reply:
#2246

Reply:
Agree. Let's fix it as part of that other issue. Would prefer to keep existing functionality as it is and not expand the scope of this PR.